File: 1.two-sum.py
#
# @lc app=leetcode id=1 lang=python3
#
# [1] Two Sum
#

# @lc code=start
import collections
class Solution:
    # HashMap, Dictionary
    # Time: O(N)
    # Space: O(N)
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        for i in range(len(nums)):
            for j in range(i + 1, len(nums)): 
                if nums[i] + nums[j] == target:
                    return [i, j]

File: 1.two-sum.cpp
/*
 * @lc app=leetcode id=1 lang=cpp
 *
 * [1] Two Sum
 */

// @lc code=start

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int n=nums.size();
        map <int , int> ma;
        for(int i=0;i<n;i++)
        {
            int num= target- nums[i];
            if (ma.count(num)!=0)
            {
                return {ma[num],i};
            }
            else ma[nums[i]]=i;
        }
        return {};
    }
};
// @lc code=end




